<!-- FroboScout launch file -->

<launch>
	<!-- FroboScout parameters --> 
	<param name="diff_steer_wheel_distance" value="0.5"/>
	<param name="ticks_per_meter_left" value="2627" />
	<param name="ticks_per_meter_right" value="2627" />

	<!-- node pkg="rosbag" name="rosbag_record" type="record" args="-a"/-->

	<group ns="fmSim">
		<node pkg="frobit_sim" name="froboscout_sim" type="frobit_sim_node.py" output="screen" >
    		<param name="deadman_sub" value="/fmCommand/deadman" />
			<param name="cmd_vel_sub" value="/fmCommand/cmd_vel" />
			<param name="pose_pub" value="/fmKnowledge/pose" />
            <param name="max_linear_acceleration" value="1.8"/>
            <param name="max_angular_acceleration" value="2.0"/>
			<param name="wheel_feedback_left_pub" value="/fmInformation/wheel_feedback_left" />
			<param name="wheel_feedback_right_pub" value="/fmInformation/wheel_feedback_right" />
			<param name="publish_wheel_feedback_rate" value="50" />
		</node>
	</group>

	<group ns="fmSensors">
	</group>

	<group ns="fmProcessors">
		<node pkg="frobit_plot" name="frobit_plot" type="frobit_plot_node.py" output="screen" >
			<param name="pose_sub" value="/fmKnowledge/pose" />
			<param name="gnss_sub" value="/fmInformation/gpgga_tranmerc" />
			<param name="odom_sub" value="/fmKnowledge/encoder_odom" />
			<param name="wptnav_status_sub" value="/fmData/wptnav_status" />
			<param name="plot_pose_track" value="true" />
			<param name="plot_gnss_track" value="false" />
			<param name="plot_odometry_track" value="false" />
			<param name="plot_pose_yaw" value="false" />
			<param name="map_trackpoint_threshold" value="0.01" />
			<param name="map_max_trackpoints" value="500" />
			<param name="map_minimum_size" value="4.0" />
			<param name="map_easting_offset" value="0.0" />
			<param name="map_northing_offset" value="0.0" />
			<param name="map_update_frequency" value="50.0" />
			<param name="map_title" value="FroboScout track" />
			<param name="map_window_size" value="6.0" />
			<param name="avatar_extension_front" value="0.195" />
			<param name="avatar_extension_rear" value="0.505" />
			<param name="avatar_extension_lateral" value="0.3" />
		</node>
	</group>

	<group ns="fmDecisionMakers">
		<node pkg="lls_surveying_2014" name="mission" type="keyboard_mission_node.py" output="screen">
			<param name="keyboard_sub" value="/fmHMI/keyboard" />
			<param name="automode_pub" value="/fmDecision/automode" />
    		<param name="deadman_pub" value="/fmCommand/deadman" />
			<param name="cmd_vel_pub" value="/fmCommand/cmd_vel" />
            <param name="max_linear_velocity" value="1.0"/>
            <param name="max_angular_velocity" value="0.4"/>
            <param name="linear_velocity_step" value="0.1"/>
            <param name="angular_velocity_step" value="0.1"/>
		</node>
	</group>

	<group ns="fmExecutors">
		<node pkg="lls_surveying_2014" type="waypoint_navigation_node.py" name="wptnav" output="screen">
			<param name="automode_sub" value="/fmDecision/automode" />
			<param name="pose_sub" value="/fmKnowledge/pose" />
			<param name="joy_sub" value="/fmLib/joy" />
			<param name="cmd_vel_pub" value="/fmCommand/cmd_vel" />
			<param name="implement_pub" value="/fmCommand/implement" />
			<param name="status_pub" value="/fmData/wptnav_status" />
			<param name="status_publish_interval" value="1" />
			<param name="print_debug_information" value="true" />

            <param name="max_linear_velocity" value="1.2"/>
            <param name="max_angular_velocity" value="0.70"/>

            <param name="drive_kp" value="1.6"/>
            <param name="drive_ki" value="0.1"/>
            <param name="drive_kd" value="0.0"/>
            <param name="drive_max_output" value="0.25"/>

            <param name="turn_kp" value="1.3"/>
            <param name="turn_ki" value="0.0"/>
            <param name="turn_kd" value="0.0"/>
            <param name="turn_max_output" value="1.0"/>

			<param name="wpt_default_mode" value="MCTE" />
			<param name="wpt_default_tolerance" value="0.30" />
			<param name="wpt_default_drive_velocity" value="1.0" />
			<param name="wpt_default_turn_velocity" value="0.7" />
			<param name="wpt_default_wait_after_arrival" value="2.0" />
			<param name="wpt_default_implement_command" value="0.0" />

			<param name="target_distance" value="1.0" />
			<param name="turn_start_at_heading_err" value="45.0" />
			<param name="turn_stop_at_heading_err" value="2.0" />
			<param name="ramp_drive_velocity_at_distance" value="0.9" />
			<param name="ramp_min_drive_velocity" value="0.15" />
			<param name="ramp_turn_velocity_at_angle" value="30.0" />
			<param name="ramp_min_turn_velocity" value="0.20" />
		</node>		
	</group>

	<group ns="fmControllers">
	</group>

	<group ns="fmLib">
	 	<node pkg="keyboard_interface" type="keyboard_node" name="keyboard">
			<param name="keyboard_pub" value="/fmHMI/keyboard" />
		</node>

		<node pkg="lls_surveying_2014" name="plot_feedback" type="plot_propulsion_feedback_node.py" output="screen" >
 			<param name="propulsion_feedback_sub" value="/fmInformation/wheel_feedback_right" />
 			<param name="update_frequency" value="50" />
			<param name="plot_title" value="Wheel feedback" />
			<param name="maximum_velocity" value="1.2" />
			<param name="maximum_thrust" value="511" />
		</node>

	</group>
</launch>

